### PR TITLE
Redo fileio with fstream

### PR DESCRIPTION
**Pull request by fundies**

### CODE REVIEW

**Comment by RobertBColton** [comment]
@time-killer-games you want to test this too please? I know you use a lot of external file handling like my games do that I tested already.

File: ENIGMAsystem/SHELL/Universal_System/fileio.cpp [hunk 1]
```diff
@@ -17,263 +17,171 @@
...
-    if (mf.spos >= mf.sdata.length()) return -1;
-  }
-  if (sscanf(mf.sdata.substr(mf.spos).c_str(),"%lf%n",&r1,&apos),apos)
-    mf.spos += apos;
-  if (mf.spos >= mf.sdata.length() && feof(mf.f))
-    mf.eof = true;
-  return r1;
+  double x;
+  enigma::files[fileid].fs >> x;
+  return x;
```
Comment (RobertBColton) [inline ENIGMAsystem/SHELL/Universal_System/fileio.cpp]: Ok here's what I want:
```cpp
if (enigma::files[fileid].fs.eof()) {
  // showError()
  // return 0;
}
```

File: ENIGMAsystem/SHELL/Universal_System/fileio.cpp [hunk 2]
```diff
@@ -17,263 +17,214 @@
...
+#define try_io_and_print(f)
+#endif
+
+namespace enigma {
+  
+  struct file {
+    file() {}
+    file(const std::string& fName, std::ios_base::openmode mode) : fn(fName) { fs.open(fName, mode); }
+    file(file&& other) : fn(other.fn) { fs.swap(other.fs); }
+    file(file& other) : fn(other.fn) { fs.swap(other.fs); }
```
Comment (fundies) [inline ENIGMAsystem/SHELL/Universal_System/fileio.cpp:41]: @RobertBColton josh says this copy constructor is wrong / unneeded but ass array won't work without it. Pls fix

Comment (RobertBColton) [reply ENIGMAsystem/SHELL/Universal_System/fileio.cpp]: Was a bug 26b6cf6cfb34542b29e1a97985ba771bd0157dc8 in OffsetVector apparently.

**Comment by time-killer-games** [comment]
what do you need tested from this pr? everything that was changed?

**Comment by RobertBColton** [comment]
I just thought you'd care because I know you use a lot of external file reading. Not everything, just games that you had using external files because he rewrote all the file text/bin functions to use the C++ classes.

**Pull request merged**